src/ape/api/transactions.py

```diff
@@ -248,27 +251,27 @@ def await_confirmations(self) -> "ReceiptAPI":
         iterations_timeout = 20
         iteration = 0
         # Wait for nonce from provider to increment.
-        if self.transaction.sender:
-            sender_nonce = self.provider.get_nonce(self.transaction.sender)
```

Comment:
One rule of thumb we've committed to in the past is that `__getattr__` methods should never be used inside of ape core directly, because when they fail, they fail quite badly (infinite loop corruption bug)

It's okay to use it for tests however

Reply:
Yep, I found that out the hard way. Also, getattr doesn't like hidden attributes to be used either, took a while to figure out what I was doing wrong